prototype1/src/app_net/helpers.rs

```diff
@@ -3,16 +3,17 @@ pub use self::helpers::{get_own_ip, get_ip};
 mod helpers {
   use std::net::{IpAddr, lookup_host, TcpStream, Shutdown};
   use std::str::FromStr;
-  use std::io::Error;
+  use std::io::{Error, ErrorKind};
 
   // TODO: Use a non-hacky solution
   pub fn get_own_ip() -> IpAddr {
-    let external_ip = host_lookup("google.com").unwrap();
-
-    let stream = TcpStream::connect((external_ip, 80)).unwrap();
-    let local_addr = stream.local_addr().unwrap();
-    let _ = stream.shutdown(Shutdown::Both);
-    local_addr.ip()
+    host_lookup("google.com")
+      .and_then(|external_ip| TcpStream::connect((external_ip, 8000)))
```

Comment:
Oops, this stayed. Point here though is that if this TcpStream fails to connect at this stage (unlikely since dns resolution would also fail) that we'll be stuck here for a while